frame/name-service/src/resolver.rs

```diff
@@ -0,0 +1,93 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Handles the mapping of a name registration to an name record.
+
+use crate::{types::*, *};
+use frame_support::traits::ReservableCurrency;
+
+impl<T: Config> NameServiceResolver<T> for Pallet<T> {
+	/// Get the native address associated with this name hash.
+	fn get_address(name_hash: NameHash) -> Option<(T::AccountId, u32)> {
+		AddressResolver::<T>::get(name_hash)
+	}
+	/// Set the native address associated with this name hash.
+	///
+	/// We allow a user to set this value without paying any additional deposits.
+	/// NOTE: do we want additional deposit here?. already deposited for the name hash itself.
```

Comment:
We already take a deposit for setting the name, which might be much larger than the address. We could make setting the address deposit-less, since most users would want to set both and seems like over depositing.

Reply:
Completely agree 👍